Schedule

vsh123 opened the pull request:
1. 스케줄 작성 프론트 구현
2. 스케줄 컨트롤러 생성
3. 프론트에 필요한 이미지 추가
![image](https://user-images.githubusercontent.com/32254689/48144772-d51ac880-e2f4-11e8-8e7f-b782632f83ae.png)

![image](https://user-images.githubusercontent.com/32254689/48144799-dea43080-e2f4-11e8-8064-4585b4f5ef40.png)

vsh123 committed:
26일 작업을 위한 토대 작성

vsh123 committed:
1. schedule front 작성
   - 일차별 탭 기능 구현
   - Spot 이동 구현
   - style 작성

vsh123 committed:
Merge remote-tracking branch 'upstream/develop' into schedule

1.conflict 수정

vsh123 committed:
1. 스케쥴 작성  href 수정
2. schedulecontroller 작성

vsh123 committed:
Merge branch 'schedule' of https://github.com/vsh123/think-jeonju into schedule

vsh123 committed:
1. conflict 수정

vsh123 committed:
Merge remote-tracking branch 'upstream/develop' into schedule

1. #10 컨플릭트 수정

vsh123 committed:
1. 스케줄 검색기능 구현
2. 스케줄 드래그앤드롭 기능 수정

vsh123 committed:
Merge remote-tracking branch 'upstream/develop' into schedule

vsh123 committed:
1. 스케줄 작성 프론트 작업
 - 지렁이(한눈에 보기) 구현
 - 지렁에이서 장소 클릭하면 정보 표현(Modal)

2. 예산 입력란 추가

hyelin0111 commented on src/main/resources/templates/schedule.html:
```diff
@@ -0,0 +1,107 @@
+                <div class="modal-body">
+                    <form name="modalForm" id="modalForm">
+                        시작일 : <input id="toDate" width="276" /><br>
+                        종료일 : <input id="fromDate" width="276" />
```
여기 width="276" 이게 적용이 되는거야??

vsh123 replied:
적용 안되는거에요 ㅠㅠㅠ 지우겠습니다!

vsh123 replied:
수정 완료!

hyelin0111 commented on src/main/java/com/kvp/thinkjeonju/controller/ApiScheduleSearchController.java:
```diff
@@ -0,0 +1,42 @@
+            spots = spotService.setLikeInSpotDTOs(m, spots);
+        }
+
+        System.out.println("Spot 찾기 완료!");
```
log로 찍는 건 어때?

vsh123 replied:
아! 이거도 테스트용도여서 지울려곸ㅋㅋㅋㅋㅋ

vsh123 replied:
수정 완료!

mindock commented on src/main/java/com/kvp/thinkjeonju/controller/ApiScheduleSearchController.java:
```diff
@@ -0,0 +1,42 @@
+@Slf4j
+@RestController
+@RequestMapping("/schedule/search")
+public class ApiScheduleSearchController {
```
ApiScheduleController 아닐까???
그래서 ApiScheduleController의 requestMapping은 /api/schedules 이게 되고
그안에 메소드로 search를 둬서 그 메소드의 mapping은 /search이런식으로 줘야할것 같은데?

vsh123 replied:
수정 완료!

mindock commented on src/main/java/com/kvp/thinkjeonju/controller/ApiScheduleSearchController.java:
```diff
@@ -0,0 +1,42 @@
+
+    @PostMapping("")
+    public ArrayList<SpotDTO> getSpotData(@RequestParam(value = "dataValue") String dataValue, HttpSession session) {
+
```
근데 schedule에서 spot을 검색하는 거면,
ApiSpotController에서 검색해야하는 거아닌가요?
ScheduleController에서 장소를 검색한다는게 이상한데
스케쥴에 뭐하는게 아니라, 단순히 장소 검색을 하는거라면, ApiSpotController를 만들어야한다고 생각합니다

mindock replied:
그리고 ResponseEntity<ArrayList<SpotDTO>> 이런식으로 상태코드랑 함께 보내줘야할 것 같음

vsh123 replied:
수정 완료!

mindock commented on src/main/java/com/kvp/thinkjeonju/controller/ScheduleController.java:
```diff
@@ -0,0 +1,21 @@
+@Controller
+@RequestMapping("/schedule")
+public class ScheduleController {
+
```
보통 url에 복수형을 많이 쓴다고 합니다.
/schedules로 바꾸는 것은 어떨까요?

vsh123 replied:
수정 완료!

mindock commented on src/main/resources/static/css/schedule.css:
```diff
@@ -0,0 +1,347 @@
+    width: 30px;
+    height: 30px;
+    /*   border: 1px solid blue;*/
+    display: inline-block;
```
안쓸거면 지우기

vsh123 replied:
넵넵

mindock commented on src/main/resources/templates/base.html:
```diff
@@ -26,7 +26,7 @@ <h1 class="logo">
                 </a>
                 {{#loginUser}}
-                <a href="#">
+                <a href="/schedule">
```
여기도 /schedules

vsh123 replied:
수정 완료!

mindock commented on src/main/resources/static/js/schedule.js:
```diff
@@ -0,0 +1,307 @@
+        var name = $(a[count]).parent().find(".spotId").attr("class").split(' ')[0]
+        var x = a[count].getAttribute("src");
+        const constHTML ="<div class=\"draw\">\n" +
+            "                        <img src=\""+ x +"\"  class=\"spotimg\">\n" +
```
근데 여기서 `\ "` 이거쓰는 이유가 `"` 이거 쓰려고 그런거야?? `'` 이거쓰면 안돼?

vsh123 replied:
수정 완료!

vsh123 committed:
1. ApiScheduleSearchController 삭제

2. ApiSpotController 수정
 - 기존의 ScheduleSearchController에 있던 메소드가 해당 컨트롤러에 더
 적합하다 판단되어  이전

3. ScheduleController 수정
 - Mapping URL 수정

4. 예산 기능 수정
 - 기존 한 장소에서 하나의 예산만 입력가능한 방식에서 현재는 한 장소당
 최대 5개의 예산을 저장할 수 있으며, 각 예산별 이름 명명이 가능
 - 이름의 최대 제한 글자 : 10자
 - 최대 적을수 있는 가격 : 9,999,999원

 5. 스케줄 작성 버그 수정
  - 2일차에서 1일차로 축소시 기존 일정이 사라지지 않는점 주성

mindock commented on src/main/java/com/kvp/thinkjeonju/controller/ApiSpotController.java:
```diff
@@ -40,4 +39,23 @@
+		else{
+			return new ResponseEntity(HttpStatus.BAD_REQUEST);
+		}
+
```
왜 login되어있지 않은 사람은 BAD_REQUEST 보내는거지??

vsh123 replied:
스케줄 작성 페이지는 로그인한 사용자만 들어갈 수 있습니다. 
하지만 만약 비로그인한 사람이 저 Schedule 내 검색(/api/spots/search)을 url 입력으로 강제적으로 들어가려고했을때 검사 결과 안나오게 하려구 저렇게 만들었습니다!

vsh123 replied:
수정 완료!!

vsh123 committed:
1. ApiSpotController 수정
 - return 위치 변경

2. schedule.css 변경
 - day-Form min-height 변경
